### PR TITLE
Add Namespaces logging to kuberentes reporter

### PR DESCRIPTION
**Pull request by ormergi**
Occasionally namespace objects deletion at test suite `AfterSuite` cleanup
fails and the object is stuck on `Terminating` phase, 
This behavior causes test suite to report error in AfterSuite and the prow-job to report failure
even if all tests passed, for example: sriov-lane jobs 
https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirt/3806/pull-kubevirt-e2e-kind-1.17-sriov/1282212196058664964

This PR extends kuberentes custom reporter to dump namespace
objects for easier debugging.

Signed-off-by: Or Mergi <ormergi@redhat.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/reporter/kubernetes.go [hunk 1]
```diff
@@ -433,6 +434,29 @@ func (r *KubernetesReporter) logSecrets(virtCli kubecli.KubevirtClient) {
...
+func (r *KubernetesReporter) logNamespaces(virtCli kubecli.KubevirtClient) {
+	f, err := os.OpenFile(filepath.Join(r.artifactsDir, fmt.Sprintf("%d_namespaces.log", r.failureCount)),
+		os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to open the file: %v\n", err)
+		return
+	}
+	defer f.Close()
+
+	nodes, err := virtCli.CoreV1().Namespaces().List(metav1.ListOptions{})
```
Comment (danielBelenky) [inline tests/reporter/kubernetes.go:446]: s/nodes/namespaces

Comment (ormergi) [reply tests/reporter/kubernetes.go]: done :)

**Comment by danielBelenky** [comment]
/lgtm
/approve
/hold

@ormergi feel free to remove the hold and merge this.

**Comment by ormergi** [comment]
> /lgtm
> /approve
> /hold
> 
> @ormergi feel free to remove the hold and merge this.

Thanks!
/unhold

**Pull request merged**